com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/util/OperationsHelper.java

```diff
@@ -59,6 +64,39 @@ private void addOngoingOperationsState(List<Operation> existingOngoingOperations
         }
     }
 
+    public void addErrorType(Operation operation) {
+        if (operation.getState() == State.ERROR) {
+            operation.setErrorType(getErrorType(operation));
+        }
+    }
+
+    public ErrorType getErrorType(Operation operation) {
+        List<HistoricOperationEvent> historicEvents = historicOperationEventService.createQuery()
+                                                                                   .processId(operation.getProcessId())
+                                                                                   .list();
+        EventType historicErrorType = null;
+        for (HistoricOperationEvent historicEvent : historicEvents) {
```

Comment:
Actually this would not achieve the desired state. In your example, if there's a CONTENT_ERROR in the list, but after it there's a RETRIED, it'd return CONTENT_ERROR even though it is retried. This is not what we want. It'd still be achievable using streams but it'd get complicated. I prefer using a 'for' for such complicated filtering.